### PR TITLE
#2366 introducing warm and flat start for model network

### PR DESCRIPTION
**Pull request by marcochiaramello**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [x] unit tests and non regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [x] if this PR modifies a dictionary: the corresponding french dictionary was updated
- [ ] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

File: dynawo/sources/Models/CPP/ModelNetwork/DYNNetworkComponent.cpp [hunk 1]
```diff
@@ -175,4 +176,17 @@ NetworkComponent::addElementWithValue(const string& elementName, const std::stri
   addSubElement("value", elementName, Element::TERMINAL, id(), parentType, elements, mapElement);
 }
 
+NetworkComponent::startingPointMode_t
+NetworkComponent::getStartingPointMode(std::string& startingPointMode) {
+  std::map<std::string, startingPointMode_t> enumResolver = {{"flat", startingPointMode_t::FLAT},
```
Comment (rosiereflo) [inline dynawo/sources/Models/CPP/ModelNetwork/DYNNetworkComponent.cpp:181]: I would put this static

**Review by rosiereflo — CHANGES_REQUESTED** [review]

**Comment by rosiereflo** [comment]
There is also 1 code smell to look at

File: dynawo/sources/Common/Dictionaries/DYNLog_en_GB.dic [hunk 2]
```diff
@@ -46,6 +46,7 @@ InternalParam                 =             parameter %1% was computed internall
 ParamUnused                   =             parameter %1% in model %2% is not used
 ParamValueInOrigin            =             parameter %1% has value %3% (coming from origin %2%)
 ModelMultiParamNotFound       =             in model multi get parameter value: Model: %1%  Parameter: %2% not found.
+StartingPointModeNotFound     =             startingPointMode %1% not found, should be warm or flat. Using warm starting point mode.
```
[the remark this answers is not in the document]
Comment (marcochiaramello) [reply dynawo/sources/Common/Dictionaries/DYNLog_en_GB.dic:49]: yes, ongoing :)

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelBus.cpp [hunk 3]
```diff
@@ -506,6 +505,18 @@ ModelBus::evalCalculatedVars() {
 
 void
 ModelBus::init(int& yNum) {
+  angle0_ = bus_->getAngle0() * DEG_TO_RAD;
```
[the remark this answers is not in the document]
Comment (marcochiaramello) [reply dynawo/sources/Models/CPP/ModelNetwork/DYNModelBus.cpp:508]: indeed, thx, done

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelGenerator.cpp [hunk 4]
```diff
@@ -42,24 +42,14 @@ namespace DYN {
...
 ModelGenerator::ModelGenerator(const shared_ptr<GeneratorInterface>& generator) :
 NetworkComponent(generator->getID()),
-stateModified_(false) {
-  // init data
-  Pc_ = -1. * generator->getP();
-  Qc_ = -1. * generator->getQ();
+generator_(generator),
+stateModified_(false),
+startingPointMode_(WARM) {
+  Pc_ = 0.;
```
[the remark this answers is not in the document]
Comment (marcochiaramello) [reply dynawo/sources/Models/CPP/ModelNetwork/DYNModelGenerator.cpp:48]: done

File: dynawo/sources/Models/CPP/ModelNetwork/DYNNetworkComponent.cpp [hunk 5]
```diff
@@ -175,4 +176,17 @@ NetworkComponent::addElementWithValue(const string& elementName, const std::stri
   addSubElement("value", elementName, Element::TERMINAL, id(), parentType, elements, mapElement);
 }
 
+NetworkComponent::startingPointMode_t
+NetworkComponent::getStartingPointMode(std::string& startingPointMode) {
```
[the remark this answers is not in the document]
Comment (marcochiaramello) [reply dynawo/sources/Models/CPP/ModelNetwork/DYNNetworkComponent.cpp:180]: thx!

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelHvdcLink.cpp [hunk 6]
```diff
@@ -505,7 +531,7 @@ ModelHvdcLink::setAttributes(const shared_ptr<HvdcLineInterface>& dcLine) {
 
 void
 ModelHvdcLink::setConvertersActivePower(const shared_ptr<HvdcLineInterface>& dcLine) {
-  if (dcLine->getConverter1()->hasP() && dcLine->getConverter2()->hasP()) {
+  if (dcLine->getConverter1()->hasP() && dcLine->getConverter2()->hasP() && startingPointMode_ == WARM) {
```
Comment (cossartque) [inline dynawo/sources/Models/CPP/ModelNetwork/DYNModelHvdcLink.cpp:534]: What will be the values of P01_ and P02_ in FLAT ?

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelHvdcLink.cpp [hunk 7]
```diff
@@ -527,7 +553,7 @@ ModelHvdcLink::setConvertersActivePower(const shared_ptr<HvdcLineInterface>& dcL
 
 void
 ModelHvdcLink::setConvertersReactivePower(const shared_ptr<HvdcLineInterface>& dcLine) {
-  if (dcLine->getConverter1()->hasQ() && dcLine->getConverter2()->hasQ()) {
+  if (dcLine->getConverter1()->hasQ() && dcLine->getConverter2()->hasQ() && startingPointMode_ == WARM) {
```
Comment (cossartque) [inline dynawo/sources/Models/CPP/ModelNetwork/DYNModelHvdcLink.cpp:556]: What will be the values of Q01_ and Q02_ in FLAT ?

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelShuntCompensator.cpp [hunk 8]
```diff
@@ -371,7 +371,24 @@ ModelShuntCompensator::evalCalculatedVarI(unsigned numCalculatedVar) const {
...
 ModelShuntCompensator::init(int& /*yNum*/) {
-  /* not needed */
+  double Q = 0.;
+  double uNode = 0.;
+  double thetaNode = shunt_->getBusInterface()->getAngle0();
+  double unomNode = shunt_->getBusInterface()->getVNom();
+  switch (startingPointMode_) {
+  case FLAT:
+    uNode = shunt_->getBusInterface()->getVNom();
+    Q = shunt_->getB(currentSection_) * vNom_ * vNom_ / SNREF;
```
Comment (cossartque) [inline dynawo/sources/Models/CPP/ModelNetwork/DYNModelShuntCompensator.cpp:381]: Do we know the currentSection_ in FLAT mode ?

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelStaticVarCompensator.cpp [hunk 9]
```diff
@@ -124,7 +101,41 @@ ModelStaticVarCompensator::setFequations(map<int, string>& /*fEquationIndex*/) {
...
+  double ui0 = 0.;
+  double U0 = 0.;
+  double P0 = svc_->getP() / SNREF;
+  double Q0;
+  double unomBus = svc_->getBusInterface()->getVNom();
+  double thetaBus0 = svc_->getBusInterface()->getAngle0();
+  double uBus0 = 0.;
+  switch (startingPointMode_) {
+  case FLAT:
+    Q0 = svc_->getReactivePowerSetPoint() / SNREF;
```
Comment (cossartque) [inline dynawo/sources/Models/CPP/ModelNetwork/DYNModelStaticVarCompensator.cpp:116]: Same convention as getQ() for getReactivePowerSetPoint() ?

File: dynawo/sources/Common/Dictionaries/DYNLog_en_GB.dic (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcochiaramello) [reply dynawo/sources/Common/Dictionaries/DYNLog_en_GB.dic:49]: agreed, done

**Review by cossartque — APPROVED** [review]

**Pull request merged**